subsys/bluetooth/common/rpa.c

```diff
@@ -14,20 +14,41 @@
 #include <stddef.h>
 #include <errno.h>
 #include <string.h>
-#include <sys/atomic.h>
-#include <sys/util.h>
-#include <sys/byteorder.h>
-#include <debug/stack.h>
-
-#include <tinycrypt/constants.h>
-#include <tinycrypt/aes.h>
-#include <tinycrypt/utils.h>
-#include <tinycrypt/cmac_mode.h>
 
 #define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_DEBUG_RPA)
 #define LOG_MODULE_NAME bt_rpa
 #include "common/log.h"
 
+#include <bluetooth/crypto.h>
+
+#if defined(CONFIG_BT_CTLR) && defined(CONFIG_BT_HOST_CRYPTO)
+#include "../controller/util/util.h"
+#include "../controller/hal/ecb.h"
+#endif /* defined(CONFIG_BT_CTLR) && defined(CONFIG_BT_HOST_CRYPTO) */
+
+static inline int _rand(void *buf, size_t len)
+{
+/* Force using controller rand function. */
+#if defined(CONFIG_BT_CTLR) && defined(CONFIG_BT_HOST_CRYPTO)
+	return util_rand(buf, len);
+#else
+	return bt_rand(buf, len);
+#endif
+}
+
+static inline int _encrypt_le(const u8_t key[16], const u8_t plaintext[16],
+			      u8_t enc_data[16])
```

Comment:
Same thing here with `inline` and the underscore prefix.

Reply:
fixed.

Reply:
@jhedberg Interesting observation: having inline avoids "warning: 'internal_rand' defined but not used"